.github/workflows/main.yml

```diff
@@ -127,21 +174,68 @@ jobs:
       run: |
         mkdir metagraph/build
         cd metagraph/build
+
+        BUILD_KMC="ON"
+        if [ "${{ matrix.build_static }}" = "ON" ]; then
+          BUILD_KMC="OFF"
+        fi
+
         cmake -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
               -DCMAKE_DBG_ALPHABET=${{ matrix.alphabet }} \
+              -DBUILD_STATIC=${{ matrix.build_static }} \
+              -DBZIP2_LIBRARY_RELEASE=/usr/local/opt/bzip2/lib/libbz2.a \
+              -DBUILD_KMC=${BUILD_KMC} \
               -DWITH_AVX=OFF -DWITH_MSSE42=OFF ..
 
     - name: build metagraph
       run: cd metagraph/build && make -j 2 metagraph
 
     - name: build unit tests
+      if: ${{ matrix.build_static == 'OFF' }}
       run: cd metagraph/build && make -j 2 unit_tests
 
     - name: build other
       run: cd metagraph/build && make -j 2
 
     - name: run unit tests
+      if: ${{ matrix.build_static == 'OFF' }}
       run: cd metagraph/build && ./unit_tests
 
     - name: run integration tests
       run: cd metagraph/build && make check
+
+    - name: rename static binary
+      if: ${{ matrix.build_static == 'ON' }}
+      run: mv metagraph/build/metagraph_${{ matrix.alphabet }} metagraph/build/metagraph_${{ matrix.alphabet }}_mac_osx
+
+    # for potential release
+    - name: upload static binary
+      if: ${{ matrix.build_static == 'ON' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: metagraph_${{ matrix.alphabet }}_static_binary_mac_osx
+        path: metagraph/build/metagraph_${{ matrix.alphabet }}_mac_osx
+
+  Release:
+    name: Create Release
+    if: contains(github.ref, 'tags/v')
+    runs-on: ubuntu-latest
+    needs: [Linux, MacOS]
+    steps:
+      - name: fetch static binary
+        uses: actions/download-artifact@v2
+        with:
+          path: artifacts
+      - name: create release with static binaries
+        uses: svenstaro/upload-release-action@v2
+        with:
+          release_name: Release ${{ github.ref }}
+          body: |
+            The binaries are currently provided on an experimental basis.
+            The Linux binaries are statically linked and hence don't require additional libraries to be installed.
+            The Mac OS X binaries dynamically link to libhts, which would need to be available on your system, e.g. using `brew install htslib`). The Mac OS X binaries are currently nor signed nor notarized - you need to explicitely allow the binary to run in the security settings of your system. 
```

Comment:
Can we link against libhts statically?

Reply:
We probably could, but libhts has some dependencies like libcurl which make it a bit harder. The strategy would probably be to build libcurl and libhts from sources (the same is actually done for linux).
But IMHO this provides already some value and installing libhts shouldn't be a big burden. My intention is to allocate my time to create official conda packages and docker images. If we still think it's useful to statically link libhts for mac, we can still do it later.

Reply:
I checked it on my laptop and found a static lib `/usr/local/opt/htslib/lib/libhts.a` (installed with brew).

I'll try to fix this, maybe it's easy.
But otherwise, yeah, we can do it later if this is not trivial.